From 826d90af0d79c485910f773901c1a74a43bb1ed2 Mon Sep 17 00:00:00 2001
From: Charles Stevens <charles.stevens@compasses.com>
Date: Tue, 1 Dec 2020 16:14:40 -0600
Subject: [PATCH 10/10] RSOM-212 - u-boot for each SOM type

Create configurations for each Beacon EmbeddedWorks SOM type
Remove unneeded code from board file I2C and WiFi initialization.

Signed-off-by: Charles Stevens <charles.stevens@compasses.com>
---
 arch/arm/dts/beacon-rzg2-baseboard.dtsi       | 356 ++++++++++++++----
 arch/arm/dts/beacon-rzg2-som.dtsi             |  23 +-
 .../dts/r8a774a1-beacon-rzg2m-kit-u-boot.dtsi |   0
 arch/arm/dts/r8a774a1-beacon-rzg2m-kit.dts    |  35 +-
 .../dts/r8a774b1-beacon-rzg2n-kit-u-boot.dtsi |  30 ++
 arch/arm/dts/r8a774b1-beacon-rzg2n-kit.dts    |  38 ++
 .../dts/r8a774e1-beacon-rzg2h-kit-u-boot.dtsi |  30 ++
 arch/arm/dts/r8a774e1-beacon-rzg2h-kit.dts    |  43 +++
 arch/arm/mach-rmobile/Kconfig.64              |  14 +-
 .../{beacon-rzg2m => beacon-rzg2}/Kconfig     |   6 +-
 board/beacon/beacon-rzg2/MAINTAINERS          |   8 +
 .../{beacon-rzg2m => beacon-rzg2}/Makefile    |   4 +-
 .../beacon-rzg2.c}                            |  20 -
 board/beacon/beacon-rzg2m/MAINTAINERS         |   6 -
 configs/r8a774a1_beacon-rzg2m_defconfig       |   4 +
 configs/r8a774b1_beacon-rzg2n_defconfig       |  71 ++++
 configs/r8a774e1_beacon-rzg2h_defconfig       |  71 ++++
 .../configs/{beacon-rzg2m.h => beacon-rzg2.h} |   0
 18 files changed, 637 insertions(+), 122 deletions(-)
 mode change 100755 => 100644 arch/arm/dts/beacon-rzg2-baseboard.dtsi
 mode change 100755 => 100644 arch/arm/dts/beacon-rzg2-som.dtsi
 mode change 100755 => 100644 arch/arm/dts/r8a774a1-beacon-rzg2m-kit-u-boot.dtsi
 mode change 100755 => 100644 arch/arm/dts/r8a774a1-beacon-rzg2m-kit.dts
 create mode 100644 arch/arm/dts/r8a774b1-beacon-rzg2n-kit-u-boot.dtsi
 create mode 100644 arch/arm/dts/r8a774b1-beacon-rzg2n-kit.dts
 create mode 100644 arch/arm/dts/r8a774e1-beacon-rzg2h-kit-u-boot.dtsi
 create mode 100644 arch/arm/dts/r8a774e1-beacon-rzg2h-kit.dts
 rename board/beacon/{beacon-rzg2m => beacon-rzg2}/Kconfig (51%)
 create mode 100644 board/beacon/beacon-rzg2/MAINTAINERS
 rename board/beacon/{beacon-rzg2m => beacon-rzg2}/Makefile (61%)
 rename board/beacon/{beacon-rzg2m/beacon-rzg2m.c => beacon-rzg2/beacon-rzg2.c} (85%)
 delete mode 100644 board/beacon/beacon-rzg2m/MAINTAINERS
 create mode 100644 configs/r8a774b1_beacon-rzg2n_defconfig
 create mode 100644 configs/r8a774e1_beacon-rzg2h_defconfig
 rename include/configs/{beacon-rzg2m.h => beacon-rzg2.h} (100%)

diff --git a/arch/arm/dts/beacon-rzg2-baseboard.dtsi b/arch/arm/dts/beacon-rzg2-baseboard.dtsi
old mode 100755
new mode 100644
index abd485406a..e24dc74ebb
--- a/arch/arm/dts/beacon-rzg2-baseboard.dtsi
+++ b/arch/arm/dts/beacon-rzg2-baseboard.dtsi
@@ -10,32 +10,22 @@
 #include <dt-bindings/clk/versaclock.h>
 
 / {
-	aliases {
-		serial0 = &scif2;
-		serial1 = &hscif0;
-		serial2 = &hscif1;
-		serial3 = &scif0;
-		serial4 = &hscif2;
-		serial5 = &scif5;
-		spi0 = &msiof0;
-		spi1 = &msiof1;
-		spi2 = &msiof2;
-		spi3 = &msiof3;
-		ethernet0 = &avb;
-	};
-
-	chosen {
-		bootargs = "ignore_loglevel rw root=/dev/nfs ip=dhcp";
-		stdout-path = "serial0:115200n8";
-	};
-
-	backlight: backlight {
+	backlight_lvds: backlight-lvds {
 		compatible = "pwm-backlight";
 		power-supply = <&reg_lcd>;
 		enable-gpios = <&gpio_exp1 3 GPIO_ACTIVE_HIGH>;
-		pwms = <&pwm0 0 50000>;
+		pwms = <&pwm2 0 25000>;
 		brightness-levels = <0 4 8 16 32 64 128 255>;
 		default-brightness-level = <6>;
+	};
+
+	backlight_dpi: backlight-dpi {
+		compatible = "pwm-backlight";
+		power-supply = <&reg_lcd>;
+		enable-gpios = <&gpio_exp1 7 GPIO_ACTIVE_LOW>;
+		pwms = <&pwm0 0 25000>;
+		brightness-levels = <0 25 33 50 63 75 88 100>;
+		default-brightness-level = <6>;
 	};	
 
 	hdmi0-out {
@@ -52,38 +42,38 @@
 	keys {
 		compatible = "gpio-keys";
 
-		key-1 {
+		key-1 { /* S19 */
 			gpios = <&gpio4 6 GPIO_ACTIVE_LOW>;
 			linux,code = <KEY_1>;
-			label = "Switch-1";
+			label = "Up";
 			wakeup-source;
 			debounce-interval = <20>;
 		};
-		key-2 {
+		key-2 { /*S20 */
 			gpios = <&gpio3 13 GPIO_ACTIVE_LOW>;
 			linux,code = <KEY_2>;
-			label = "Switch-2";
+			label = "Left";
 			wakeup-source;
 			debounce-interval = <20>;
 		};
-		key-3 {
+		key-3 { /* S21 */
 			gpios = <&gpio5 17 GPIO_ACTIVE_LOW>;
 			linux,code = <KEY_3>;
-			label = "Switch-3";
+			label = "Down";
 			wakeup-source;
 			debounce-interval = <20>;
 		};
-		key-4 {
+		key-4 { /* S22 */
 			gpios = <&gpio5 20 GPIO_ACTIVE_LOW>;
 			linux,code = <KEY_4>;
-			label = "Switch-4";
+			label = "Right";
 			wakeup-source;
 			debounce-interval = <20>;
 		};
-		key-5 {
+		key-5 { /* S23 */
 			gpios = <&gpio5 22 GPIO_ACTIVE_LOW>;
 			linux,code = <KEY_5>;
-			label = "Switch-4";
+			label = "Center";
 			wakeup-source;
 			debounce-interval = <20>;
 		};
@@ -102,17 +92,58 @@
 		led1 {
 			gpios = <&gpio7 0 GPIO_ACTIVE_HIGH>;
 			label = "LED1";
-			linux,default-trigger = "heartbeat";
 		};
 		led2 {
 			gpios = <&gpio7 1 GPIO_ACTIVE_HIGH>;
 			label = "LED2";
-			linux,default-trigger = "heartbeat";
 		};
 		led3 {
 			gpios = <&gpio7 3 GPIO_ACTIVE_HIGH>;
 			label = "LED3";
-			linux,default-trigger = "heartbeat";
+		};
+	};
+
+	lvds {
+		compatible = "panel-lvds";
+		power-supply = <&reg_lcd_reset>;
+		width-mm = <223>;
+		height-mm = <125>;
+		backlight = <&backlight_lvds>;
+		data-mapping = "vesa-24";
+
+		panel-timing {
+			/* 800x480@60Hz */
+			clock-frequency = <30000000>;
+			hactive = <800>;
+			vactive = <480>;
+			hsync-len = <48>;
+			hfront-porch = <40>;
+			hback-porch = <40>;
+			vfront-porch = <13>;
+			vback-porch = <29>;
+			vsync-len = <1>;
+			hsync-active = <1>;
+			vsync-active = <3>;
+			de-active = <1>;
+			pixelclk-active = <0>;
+		};
+
+		port {
+			panel_in: endpoint {
+				remote-endpoint = <&lvds0_out>;
+			};
+		};
+	};
+
+	rgb {
+		/* Some random LCD with similar timings */
+		compatible = "rocktech,rk070er9427";
+		backlight = <&backlight_dpi>;
+		enable-gpios = <&gpio1 21 GPIO_ACTIVE_HIGH>;
+		port {
+			rgb_panel: endpoint {
+				remote-endpoint = <&du_out_rgb>;
+			};
 		};
 	};
 
@@ -121,7 +152,7 @@
 		regulator-name = "audio-1.8V";
 		regulator-min-microvolt = <1800000>;
 		regulator-max-microvolt = <1800000>;
-		gpio = <&gpio_exp2 7 GPIO_ACTIVE_HIGH>;
+		gpio = <&gpio_exp4 1 GPIO_ACTIVE_HIGH>;
 		enable-active-high;
 	};
 	
@@ -196,11 +227,16 @@
 };
 
 &audio_clk_a {
-	clock-frequency = <22579200>;
+	/delete-property/ clock-frequency;
+	#clock-cells = <0>;
+	compatible = "fixed-factor-clock";
+	clock-mult = <1>;
+	clock-div = <1>;
+	clocks = <&versaclock6_bb 4>;
 };
 
 &audio_clk_b {
-	clock-frequency = <22579200>;
+	clock-frequency = <22579200>; /* Support 44.1KHz */
 };
 
 &can0 {
@@ -217,32 +253,51 @@
 	status = "okay";
 };
 
-&du {
-	pinctrl-0 = <&du_pins>;
-	pinctrl-names = "default";
-	status = "okay";
+&csi20 {
+	status = "disabled"; /* Enable when we have cameras */
 
-	clocks = <&cpg CPG_MOD 724>,
-		<&cpg CPG_MOD 723>,
-		<&cpg CPG_MOD 722>,
-		<&versaclock5 1>,
-		<&x302_clk>,
-		<&versaclock5 2>;
-	clock-names = "du.0", "du.1", "du.2",
-		"dclkin.0", "dclkin.1", "dclkin.2";
+	ports {
+		port@0 {
+			reg = <0>;
+			csi20_in: endpoint {
+				clock-lanes = <0>;
+				data-lanes = <1 2>;
+				/* remote-endpoint = <&ov5640_to_mipi_csi2>; */
+			};
+		};
+	};
+};
+
+&csi40 {
+	status = "disabled";  /* Enable when we have cameras */
+
+	ports {
+		port@0 {
+			reg = <0>;
+			csi40_in: endpoint {
+				clock-lanes = <0>;
+				data-lanes = <1 2 3 4>;
+				/* remote-endpoint = <&ov5640_2_to_mipi_csi2>; */
+			};
+		};
+	};
+};
+
+&du_out_rgb {
+	remote-endpoint = <&rgb_panel>;
 };
 
 &ehci0 {
 	dr_mode = "otg";
 	status = "okay";
-	clocks = <&cpg CPG_MOD 703>, <&cpg CPG_MOD 704>, <&versaclock5 3>, <&versaclock6_bb 4>;
+	clocks = <&cpg CPG_MOD 703>, <&cpg CPG_MOD 704>, <&versaclock5 3>;
 };
 
 &ehci1 {
 	status = "okay";
-	/* Hack, we are asking the EHCI to take a clk ref to the Ethernet Phy
-       refclock so Ethernet will work */
-	clocks = <&cpg CPG_MOD 703>, <&cpg CPG_MOD 704>, <&versaclock5 4>;
+	/* Hack, we are asking the EHCI to take a clk ref to the Ethernet
+	refclock so Ethernet will work (AVB_TXCREFCLK) */
+	clocks = <&cpg CPG_MOD 703>, <&cpg CPG_MOD 704>, <&versaclock5 3>, <&versaclock5 4>;
 };
 
 &hdmi0 {
@@ -287,11 +342,21 @@
 
 &i2c2 {
 	status = "okay";
-	clock-frequency = <100000>;
+	clock-frequency = <400000>;
 
 	pinctrl-0 = <&i2c2_pins>;
 	pinctrl-names = "default";
 
+	/* Eventual Home of CSI0 Camera */
+
+	/* 0 - CSI0_RSTn */
+	/* 1 - CAM0_PW_EN_0 */
+	/* 2 - CAM0_PW_EN_1 */
+	/* 3 - CSI1_RSTn */
+	/* 4 - CAM1_PW_EN_0 */
+	/* 5 - CAM1_PW_EN_1 */
+	/* 6 - nc */
+	/* 7 - nc */
 	gpio_exp2: gpio@21 {
 		compatible = "onnn,pca9654";
 		reg = <0x21>;
@@ -299,6 +364,14 @@
 		#gpio-cells = <2>;
 	};
 
+	/* 0 - PCIECLKREQn */
+	/* 1 - PCIEWAKEn */
+	/* 2 - W_DISABLE1n */
+	/* 3 - CAT6_POWER_OFFn */
+	/* 4 - PEWAKEn */
+	/* 5 - PM_ADBUS4 */
+	/* 6 - CLKREQn */
+	/* 7 - SSD_RESETn */
 	gpio_exp3: gpio@22 {
 		compatible = "onnn,pca9654";
 		reg = <0x22>;
@@ -306,15 +379,31 @@
 		#gpio-cells = <2>;
 	};
 	
+	/* 0 - SSD_DEV_SLP */
+	/* 1 - CODEC_SPK_PWR_EN */
+	/* 2 - nc */
+	/* 3 - USBC-ENn_MUX */
+	/* 4 - CELLMOD_RESETn */
+	/* 5 - nc */
+	/* 6 - nc */
+	/* 7 - nc */
+	gpio_exp4: gpio@23 {
+		compatible = "onnn,pca9654";
+		reg = <0x23>;
+		gpio-controller;
+		#gpio-cells = <2>;
+	};
+
 	versaclock6_bb: versaclock6_bb@6a {
 		compatible = "idt,5p49v6965";
 		reg = <0x6a>;
 		#clock-cells = <1>;
 		clocks = <&x304_clk>;
 		clock-names = "xin";
-		/* CSI0_MCLK, CSI1_MCLK, AUDIO_CLKIN, USB_HUB_MCLK_BB */
+		/* CSI0_MCLK, CSI1_MCLK, AUDIO_CLK_CODEC, AUDIO_CLKA */
+		clock-output-names = "versaclock6_bb.out0_sel_i2cb", "versaclock6_bb.out1", "versaclock6_bb.out2", "versaclock6_bb.out3", "versaclock6_bb.out4";
 		assigned-clocks = <&versaclock6_bb 1>, <&versaclock6_bb 2>, <&versaclock6_bb 3>, <&versaclock6_bb 4>;
-		assigned-clock-rates =	<24000000>, <24000000>, <24000000>, <24000000>;
+		assigned-clock-rates =	<24000000>, <24000000>, <24000000>, <24576000>;
 
 		OUT1 {
 			idt,mode = <VC5_CMOS>;
@@ -330,7 +419,7 @@
 
 		OUT3 {
 			idt,mode = <VC5_CMOS>;
-			idt,voltage-microvolts = <1800000>;
+			idt,voltage-microvolts = <3300000>;
 			idt,slew-percent = <100>;
 		};
 
@@ -352,13 +441,17 @@
 
 &i2c5 {
 	status = "okay";
-	clock-frequency = <100000>;
+	clock-frequency = <400000>;
 	pinctrl-0 = <&i2c5_pins>;
 	pinctrl-names = "default";
 
+	/* Eventual Home of CSI1 Camera */
+
 	codec: wm8962@1a {
 		compatible = "wlf,wm8962";
 		reg = <0x1a>;
+		clocks = <&versaclock6_bb 3>;
+		clock-names = "mclk";
 		DCVDD-supply = <&reg_audio>;
 		DBVDD-supply = <&reg_audio>;
 		AVDD-supply = <&reg_audio>;
@@ -389,6 +482,7 @@
 	/* 4 - Touch_shdwn */
 	/* 5 - LCD_H_pol */
 	/* 6 - lcd_V_pol */
+	/* 7 - BL_Disable */
 	gpio_exp1: gpio@20 {
 		compatible = "onnn,pca9654";
 		reg = <0x20>;
@@ -403,6 +497,44 @@
 		interrupts = <9 IRQ_TYPE_EDGE_RISING>;
 		wakeup-source;
 	};
+
+	hd3ss3220@47 {
+		compatible = "ti,hd3ss3220";
+		reg = <0x47>;
+		interrupt-parent = <&gpio6>;
+		interrupts = <4 IRQ_TYPE_LEVEL_LOW>;
+
+		connector {
+			compatible = "usb-c-connector";
+			label = "USB-C";
+			data-role = "dual";
+
+			ports {
+				#address-cells = <1>;
+				#size-cells = <0>;
+
+				port@1 {
+					reg = <1>;
+					hd3ss3220_ep: endpoint {
+						remote-endpoint = <&usb3_role_switch>;
+					};
+				};
+			};
+		};
+	};
+
+};
+
+&lvds0 {
+	status = "okay";
+
+	ports {
+		port@1 {
+			lvds0_out: endpoint {
+				remote-endpoint = <&panel_in>;
+			};
+		};
+	};
 };
 
 &ohci0 {
@@ -427,6 +559,11 @@
 };
 
 &pfc {
+	scif4_pins: scif4 {
+		groups = "scif4_data_c";
+		function = "scif4";
+	};
+
 	can0_pins: can0 {
 		groups = "can0_data_a";
 		function = "can0";
@@ -468,6 +605,11 @@
 		function = "pwm0";
 	};
 
+	pwm2_pins: pwm2 {
+		groups = "pwm2_a";
+		function = "pwm2";
+	};
+
 	sdhi0_pins: sd0 {
 		groups = "sdhi0_data4", "sdhi0_ctrl";
 		function = "sdhi0";
@@ -486,7 +628,7 @@
 	};
 
 	sound_clk_pins: sound_clk {
-		groups = "audio_clk_a_a";
+		groups = "audio_clk_a_a", "audio_clk_b_a";
 		function = "audio_clk";
 	};
 
@@ -518,6 +660,12 @@
 	status = "okay";
 };
 
+&pwm2 {
+	pinctrl-0 = <&pwm2_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+};
+
 &rcar_sound {
 	pinctrl-0 = <&sound_pins &sound_clk_pins>;
 	pinctrl-names = "default";
@@ -531,23 +679,6 @@
 
 	status = "okay";
 
-	clocks = <&cpg CPG_MOD 1005>,
-		 <&cpg CPG_MOD 1006>, <&cpg CPG_MOD 1007>,
-		 <&cpg CPG_MOD 1008>, <&cpg CPG_MOD 1009>,
-		 <&cpg CPG_MOD 1010>, <&cpg CPG_MOD 1011>,
-		 <&cpg CPG_MOD 1012>, <&cpg CPG_MOD 1013>,
-		 <&cpg CPG_MOD 1014>, <&cpg CPG_MOD 1015>,
-		 <&cpg CPG_MOD 1022>, <&cpg CPG_MOD 1023>,
-		 <&cpg CPG_MOD 1024>, <&cpg CPG_MOD 1025>,
-		 <&cpg CPG_MOD 1026>, <&cpg CPG_MOD 1027>,
-		 <&cpg CPG_MOD 1028>, <&cpg CPG_MOD 1029>,
-		 <&cpg CPG_MOD 1030>, <&cpg CPG_MOD 1031>,
-		 <&cpg CPG_MOD 1020>, <&cpg CPG_MOD 1021>,
-		 <&cpg CPG_MOD 1020>, <&cpg CPG_MOD 1021>,
-		 <&cpg CPG_MOD 1019>, <&cpg CPG_MOD 1018>,
-		 <&audio_clk_a>, <&audio_clk_b>, <&audio_clk_c>, 
-		 <&cpg CPG_CORE R8A774A1_CLK_S0D4>;
-
 	ports {
         #address-cells = <1>;
         #size-cells = <0>;
@@ -585,13 +716,18 @@
 	status = "okay";
 	cs-gpios = <&gpio3 10 GPIO_ACTIVE_LOW>;
 
-	spidev1@0x00 {
+	spidev1@0 {
 	       compatible = "spidev";
 	       spi-max-frequency = <5000000>;
 	       reg = <0>;
 	};
 };
 
+&rwdt {
+	status = "okay";
+	timeout-sec = <60>;
+};
+
 &scif0 {
 	pinctrl-0 = <&scif0_pins>;
 	pinctrl-names = "default";
@@ -599,6 +735,13 @@
 	status = "okay";
 };
 
+&scif4 {
+	pinctrl-0 = <&scif4_pins>;
+	pinctrl-names = "default";
+
+	status = "okay";
+};
+
 &scif5 {
 	pinctrl-0 = <&scif5_pins>;
 	pinctrl-names = "default";
@@ -628,6 +771,26 @@
 	shared-pin;
 };
 
+&tmu0 {
+	status = "okay";
+};
+
+&tmu1 {
+	status = "okay";
+};
+
+&tmu2 {
+	status = "okay";
+};
+
+&tmu3 {
+	status = "okay";
+};
+
+&tmu4 {
+	status = "okay";
+};
+
 &usb2_phy0 {
 	pinctrl-0 = <&usb0_pins>;
 	pinctrl-names = "default";
@@ -641,18 +804,53 @@
 };
 
 &usb3_peri0 {
-	phys = <&usb3_phy0>;
-	phy-names = "usb";
+	companion = <&xhci0>;
 	status = "okay";
+	usb-role-switch;
+
+	port {
+		usb3_role_switch: endpoint {
+			remote-endpoint = <&hd3ss3220_ep>;
+		};
+	};
 };
 
 &usb3_phy0 {
 	status = "okay";
 };
 
+&vin0 {
+	status = "okay";
+};
+&vin1 {
+	status = "okay";
+};
+&vin2 {
+	status = "okay";
+};
+&vin3 {
+	status = "okay";
+};
+&vin4 {
+	status = "okay";
+};
+&vin5 {
+	status = "okay";
+};
+&vin6 {
+	status = "okay";
+};
+&vin7 {
+	status = "okay";
+};
+
 &xhci0
 {
 	pinctrl-0 = <&usb30_pins>;
 	pinctrl-names = "default";
 	status = "okay";
 };
+
+&msiof_ref_clk {
+	clock-frequency = <50000000>;
+};
diff --git a/arch/arm/dts/beacon-rzg2-som.dtsi b/arch/arm/dts/beacon-rzg2-som.dtsi
old mode 100755
new mode 100644
index fa642028d4..7c88a0b78b
--- a/arch/arm/dts/beacon-rzg2-som.dtsi
+++ b/arch/arm/dts/beacon-rzg2-som.dtsi
@@ -15,11 +15,6 @@
 		reg = <0x0 0x48000000 0x0 0x78000000>;
 	};
 
-	memory@600000000 {
-		device_type = "memory";
-		reg = <0x6 0x00000000 0x0 0x80000000>;
-	};
-
 	reserved-memory {
 		#address-cells = <2>;
 		#size-cells = <2>;
@@ -124,6 +119,7 @@
 		device-wakeup-gpios = <&pca9654 5 GPIO_ACTIVE_HIGH>;
 		clocks = <&osc_32k>;
 		clock-names = "extclk";
+		max-speed = <4000000>;
 	};
 };
 
@@ -220,13 +216,11 @@
 	status = "okay";
 };
 
-#if 01
 &hscif2 {
 	status = "okay";
 	pinctrl-0 = <&hscif2_pins>;
 	pinctrl-names = "default";
 };
-#endif
 
 &gpio6 {
 
@@ -278,7 +272,7 @@
 
 &i2c4 {
 	status = "okay";
-	clock-frequency = <400000>;
+	clock-frequency = <100000>;
 
 	pca9654: gpio@20 {
 		compatible = "onnn,pca9654";
@@ -364,8 +358,17 @@
 	};
 };
 
-&vspb {
-	status = "okay";
+/* Use explicit path to node to enable since SoC dts files
+ *  seem to have different node names for this node.
+    M and N refer to this peripheral as &vspb while the H
+    calls it &vspd.
+ */
+/ {
+	soc {
+		vspm@fe960000 {
+			status = "okay";
+		};
+	};
 };
 
 &vspi0 {
diff --git a/arch/arm/dts/r8a774a1-beacon-rzg2m-kit-u-boot.dtsi b/arch/arm/dts/r8a774a1-beacon-rzg2m-kit-u-boot.dtsi
old mode 100755
new mode 100644
diff --git a/arch/arm/dts/r8a774a1-beacon-rzg2m-kit.dts b/arch/arm/dts/r8a774a1-beacon-rzg2m-kit.dts
old mode 100755
new mode 100644
index 9c19c9d19f..ef564f5638
--- a/arch/arm/dts/r8a774a1-beacon-rzg2m-kit.dts
+++ b/arch/arm/dts/r8a774a1-beacon-rzg2m-kit.dts
@@ -5,6 +5,39 @@
 #include "beacon-rzg2-baseboard.dtsi"
 
 / {
-	model = "Beacon Embedded Works RZ/G2M Development Kit";
+	model = "Beacon EmbeddedWorks RZ/G2M Development Kit";
 	compatible =	"beacon,beacon-rzg2m", "renesas,r8a774a1";
+
+	aliases {
+		serial0 = &scif2;
+		serial1 = &hscif0;
+		serial2 = &hscif1;
+		serial3 = &scif0;
+		serial4 = &hscif2;
+		serial5 = &scif5;
+		serial6 = &scif4;
+		ethernet0 = &avb;
+	};
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+	};
+
+	memory@600000000 {
+		device_type = "memory";
+		reg = <0x6 0x00000000 0x0 0x80000000>;
+	};
+};
+
+&du {
+	status = "okay";
+
+	clocks = <&cpg CPG_MOD 724>,
+		<&cpg CPG_MOD 723>,
+		<&cpg CPG_MOD 722>,
+		<&versaclock5 1>,
+		<&x302_clk>,
+		<&versaclock5 2>;
+	clock-names = "du.0","du.1", "du.2",
+		"dclkin.0", "dclkin.1", "dclkin.2";
 };
diff --git a/arch/arm/dts/r8a774b1-beacon-rzg2n-kit-u-boot.dtsi b/arch/arm/dts/r8a774b1-beacon-rzg2n-kit-u-boot.dtsi
new file mode 100644
index 0000000000..71fb06d3e7
--- /dev/null
+++ b/arch/arm/dts/r8a774b1-beacon-rzg2n-kit-u-boot.dtsi
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Copyright 2020 Compass Electronics Group, LLC
+ */
+
+/ {
+	soc {
+		u-boot,dm-pre-reloc;
+	};
+};
+
+&cpg {
+	u-boot,dm-pre-reloc;
+};
+
+&extal_clk {
+	u-boot,dm-pre-reloc;
+};
+
+&prr {
+	u-boot,dm-pre-reloc;
+};
+
+&extalr_clk {
+	u-boot,dm-pre-reloc;
+};
+
+&sdhi2 {
+	status = "disabled";
+};
diff --git a/arch/arm/dts/r8a774b1-beacon-rzg2n-kit.dts b/arch/arm/dts/r8a774b1-beacon-rzg2n-kit.dts
new file mode 100644
index 0000000000..5050b73b3e
--- /dev/null
+++ b/arch/arm/dts/r8a774b1-beacon-rzg2n-kit.dts
@@ -0,0 +1,38 @@
+/dts-v1/;
+
+#include "r8a774b1.dtsi"
+#include "beacon-rzg2-som.dtsi"
+#include "beacon-rzg2-baseboard.dtsi"
+
+/ {
+	model = "Beacon Embedded Works RZ/G2N Development Kit";
+	compatible =	"beacon,beacon-rzg2n", "renesas,r8a774b1";
+
+	aliases {
+		serial0 = &scif2;
+		serial1 = &hscif0;
+		serial2 = &hscif1;
+		serial3 = &scif0;
+		serial4 = &hscif2;
+		serial5 = &scif5;
+		serial6 = &scif4;
+		ethernet0 = &avb;
+	};
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+	};
+};
+
+&du {
+	status = "okay";
+
+	clocks = <&cpg CPG_MOD 724>,
+		<&cpg CPG_MOD 723>,
+		<&cpg CPG_MOD 721>,
+		<&versaclock5 1>,
+		<&x302_clk>,
+		<&versaclock5 2>;
+	clock-names = "du.0", "du.1", "du.3",
+		"dclkin.0", "dclkin.1", "dclkin.3";
+};
diff --git a/arch/arm/dts/r8a774e1-beacon-rzg2h-kit-u-boot.dtsi b/arch/arm/dts/r8a774e1-beacon-rzg2h-kit-u-boot.dtsi
new file mode 100644
index 0000000000..71fb06d3e7
--- /dev/null
+++ b/arch/arm/dts/r8a774e1-beacon-rzg2h-kit-u-boot.dtsi
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Copyright 2020 Compass Electronics Group, LLC
+ */
+
+/ {
+	soc {
+		u-boot,dm-pre-reloc;
+	};
+};
+
+&cpg {
+	u-boot,dm-pre-reloc;
+};
+
+&extal_clk {
+	u-boot,dm-pre-reloc;
+};
+
+&prr {
+	u-boot,dm-pre-reloc;
+};
+
+&extalr_clk {
+	u-boot,dm-pre-reloc;
+};
+
+&sdhi2 {
+	status = "disabled";
+};
diff --git a/arch/arm/dts/r8a774e1-beacon-rzg2h-kit.dts b/arch/arm/dts/r8a774e1-beacon-rzg2h-kit.dts
new file mode 100644
index 0000000000..59c9da9288
--- /dev/null
+++ b/arch/arm/dts/r8a774e1-beacon-rzg2h-kit.dts
@@ -0,0 +1,43 @@
+/dts-v1/;
+
+#include "r8a774e1.dtsi"
+#include "beacon-rzg2-som.dtsi"
+#include "beacon-rzg2-baseboard.dtsi"
+
+/ {
+	model = "Beacon Embedded Works RZ/G2H Development Kit";
+	compatible =	"beacon,beacon-rzg2h", "renesas,r8a774e1";
+
+	aliases {
+		serial0 = &scif2;
+		serial1 = &hscif0;
+		serial2 = &hscif1;
+		serial3 = &scif0;
+		serial4 = &hscif2;
+		serial5 = &scif5;
+		serial6 = &scif4;
+		ethernet0 = &avb;
+	};
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+	};
+
+	memory@500000000 {
+		device_type = "memory";
+		reg = <0x5 0x00000000 0x0 0x80000000>;
+	};
+};
+
+&du {
+	status = "okay";
+
+	clocks = <&cpg CPG_MOD 724>,
+		<&cpg CPG_MOD 723>,
+		<&cpg CPG_MOD 721>,
+		<&versaclock5 1>,
+		<&x302_clk>,
+		<&versaclock5 2>;
+	clock-names = "du.0", "du.1", "du.3",
+		"dclkin.0", "dclkin.1", "dclkin.3";
+};
diff --git a/arch/arm/mach-rmobile/Kconfig.64 b/arch/arm/mach-rmobile/Kconfig.64
index df61340c23..7a9bf1ad7d 100644
--- a/arch/arm/mach-rmobile/Kconfig.64
+++ b/arch/arm/mach-rmobile/Kconfig.64
@@ -72,6 +72,18 @@ config TARGET_BEACON_RZG2M
           Support for Beacon RZG2M platform
 	select BOARD_LATE_INIT
 
+config TARGET_BEACON_RZG2N
+	bool "Beacon Embedded Works RZ/G2N board"
+	help
+          Support for Beacon RZG2N platform
+	select BOARD_LATE_INIT
+
+config TARGET_BEACON_RZG2H
+	bool "Beacon Embedded Works RZ/G2H board"
+	help
+          Support for Beacon RZG2H platform
+	select BOARD_LATE_INIT
+
 config TARGET_HIHOPE_RZG2M
 	bool "HiHope RZ/G2M board"
 	help
@@ -98,7 +110,7 @@ source "board/renesas/ebisu/Kconfig"
 source "board/renesas/salvator-x/Kconfig"
 source "board/renesas/ulcb/Kconfig"
 source "board/renesas/ek874/Kconfig"
-source "board/beacon/beacon-rzg2m/Kconfig"
+source "board/beacon/beacon-rzg2/Kconfig"
 source "board/renesas/hihope-rzg2m/Kconfig"
 source "board/renesas/hihope-rzg2n/Kconfig"
 source "board/renesas/hihope-rzg2h/Kconfig"
diff --git a/board/beacon/beacon-rzg2m/Kconfig b/board/beacon/beacon-rzg2/Kconfig
similarity index 51%
rename from board/beacon/beacon-rzg2m/Kconfig
rename to board/beacon/beacon-rzg2/Kconfig
index c03857cf2f..1d92763f02 100644
--- a/board/beacon/beacon-rzg2m/Kconfig
+++ b/board/beacon/beacon-rzg2/Kconfig
@@ -1,15 +1,15 @@
-if TARGET_BEACON_RZG2M
+if TARGET_BEACON_RZG2M || TARGET_BEACON_RZG2N || TARGET_BEACON_RZG2H
 
 config SYS_SOC
 	default "rmobile"
 
 config SYS_BOARD
-	default "beacon-rzg2m"
+	default "beacon-rzg2"
 
 config SYS_VENDOR
 	default "beacon"
 
 config SYS_CONFIG_NAME
-	default "beacon-rzg2m"
+	default "beacon-rzg2"
 
 endif
diff --git a/board/beacon/beacon-rzg2/MAINTAINERS b/board/beacon/beacon-rzg2/MAINTAINERS
new file mode 100644
index 0000000000..9623e1f177
--- /dev/null
+++ b/board/beacon/beacon-rzg2/MAINTAINERS
@@ -0,0 +1,8 @@
+BEACON_RZG2 SERIES BOARDS
+M:	Adam Ford <aford173@gmail.com>
+S:	Maintained
+F:	board/renesas/beacon-rzg2/
+F:	include/configs/beacon-rzg2.h
+F:	configs/r8a774a1_beacon-rzg2m_defconfig
+F:	configs/r8a774b1_beacon-rzg2n_defconfig
+F:	configs/r8a774e1_beacon-rzg2h_defconfig
diff --git a/board/beacon/beacon-rzg2m/Makefile b/board/beacon/beacon-rzg2/Makefile
similarity index 61%
rename from board/beacon/beacon-rzg2m/Makefile
rename to board/beacon/beacon-rzg2/Makefile
index 439f3199aa..5d25716af3 100644
--- a/board/beacon/beacon-rzg2m/Makefile
+++ b/board/beacon/beacon-rzg2/Makefile
@@ -1,9 +1,9 @@
 #
-# board/renesas/hihope-rzg2m/Makefile
+# board/beacon/beacon-rzg2/Makefile
 #
 # Copyright (C) 2019 Renesas Electronics Corporation
 #
 # SPDX-License-Identifier: GPL-2.0+
 #
 
-obj-y	:= beacon-rzg2m.o
+obj-y	:= beacon-rzg2.o
diff --git a/board/beacon/beacon-rzg2m/beacon-rzg2m.c b/board/beacon/beacon-rzg2/beacon-rzg2.c
similarity index 85%
rename from board/beacon/beacon-rzg2m/beacon-rzg2m.c
rename to board/beacon/beacon-rzg2/beacon-rzg2.c
index 3040849df5..bf3d407ba8 100644
--- a/board/beacon/beacon-rzg2m/beacon-rzg2m.c
+++ b/board/beacon/beacon-rzg2/beacon-rzg2.c
@@ -31,17 +31,6 @@ void s_init(void)
 {
 }
 
-#define GPIO_WLAN_REG_ON 157
-#define GPIO_BT_REG_ON 158
-
-void clear_wlan_bt_reg_on(void)
-{
-	gpio_request(GPIO_WLAN_REG_ON, "wlan_reg_on");
-	gpio_request(GPIO_BT_REG_ON, "bt_reg_on");
-	gpio_direction_output(GPIO_WLAN_REG_ON, 0);
-	gpio_direction_output(GPIO_BT_REG_ON, 0);
-}
-
 #define SCIF2_MSTP310		BIT(10)	/* SCIF2 */
 #define DVFS_MSTP926		BIT(26)
 #define GPIO2_MSTP910		BIT(10)
@@ -51,10 +40,6 @@ void clear_wlan_bt_reg_on(void)
 
 int board_early_init_f(void)
 {
-#if defined(CONFIG_SYS_I2C) && defined(CONFIG_SYS_I2C_SH)
-	/* DVFS for reset */
-	mstp_clrbits_le32(SMSTPCR9, SMSTPCR9, DVFS_MSTP926);
-#endif
 	return 0;
 }
 
@@ -81,7 +66,6 @@ int board_init(void)
 	/* low power status */
 	setbits_le16(HSUSB_REG_LPSTS, HSUSB_REG_LPSTS_SUSPM_NORMAL);
 
-	clear_wlan_bt_reg_on();
 	return 0;
 }
 
@@ -129,12 +113,8 @@ int dram_init_banksize(void)
 
 void reset_cpu(ulong addr)
 {
-#if defined(CONFIG_SYS_I2C) && defined(CONFIG_SYS_I2C_SH)
-	i2c_reg_write(CONFIG_SYS_I2C_POWERIC_ADDR, 0x20, 0x80);
-#else
 	/* only CA57 ? */
 	writel(RST_CODE, RST_CA57RESCNT);
-#endif
 }
 
 void board_add_ram_info(int use_default)
diff --git a/board/beacon/beacon-rzg2m/MAINTAINERS b/board/beacon/beacon-rzg2m/MAINTAINERS
deleted file mode 100644
index d70a524e9d..0000000000
--- a/board/beacon/beacon-rzg2m/MAINTAINERS
+++ /dev/null
@@ -1,6 +0,0 @@
-BEACON_RZG2M BOARD
-M:	Nobuhiro Iwamatsu <iwamatsu@nigauri.org>
-S:	Maintained
-F:	board/renesas/beacon-rzg2m/
-F:	include/configs/beacon-rzg2m.h
-F:	configs/r8a774a1_beacon-rzg2m_defconfig
diff --git a/configs/r8a774a1_beacon-rzg2m_defconfig b/configs/r8a774a1_beacon-rzg2m_defconfig
index 9ec037e382..5df53d4170 100644
--- a/configs/r8a774a1_beacon-rzg2m_defconfig
+++ b/configs/r8a774a1_beacon-rzg2m_defconfig
@@ -59,9 +59,13 @@ CONFIG_DM_REGULATOR_GPIO=y
 CONFIG_SCIF_CONSOLE=y
 CONFIG_USB=y
 CONFIG_DM_USB=y
+CONFIG_WDT=y
+CONFIG_HW_WATCHDOG=y
+CONFIG_WDT_RENESAS=y
 CONFIG_USB_XHCI_HCD=y
 CONFIG_USB_EHCI_HCD=y
 CONFIG_USB_EHCI_GENERIC=y
 CONFIG_USB_STORAGE=y
 CONFIG_OF_LIBFDT_OVERLAY=y
 CONFIG_SMBIOS_MANUFACTURER=""
+CONFIG_PRAM=1
diff --git a/configs/r8a774b1_beacon-rzg2n_defconfig b/configs/r8a774b1_beacon-rzg2n_defconfig
new file mode 100644
index 0000000000..dc879d68f8
--- /dev/null
+++ b/configs/r8a774b1_beacon-rzg2n_defconfig
@@ -0,0 +1,71 @@
+CONFIG_ARM=y
+CONFIG_ARCH_RMOBILE=y
+CONFIG_SYS_TEXT_BASE=0x50000000
+CONFIG_SYS_MALLOC_F_LEN=0x2000
+CONFIG_RCAR_GEN3=y
+CONFIG_R8A774B1=y
+CONFIG_TARGET_BEACON_RZG2N=y
+CONFIG_SMBIOS_PRODUCT_NAME=""
+CONFIG_FIT=y
+# CONFIG_ARCH_FIXUP_FDT_MEMORY is not set
+CONFIG_USE_BOOTARGS=y
+CONFIG_BOOTARGS="root=/dev/nfs rw nfsroot=192.168.0.1:/export/rfs ip=192.168.0.20"
+CONFIG_SUPPORT_RAW_INITRD=y
+CONFIG_DEFAULT_FDT_FILE="r8a774b1-beacon-rzg2n-kit.dtb"
+CONFIG_VERSION_VARIABLE=y
+CONFIG_HUSH_PARSER=y
+CONFIG_CMD_BOOTZ=y
+CONFIG_CMD_GPIO=y
+CONFIG_CMD_I2C=y
+CONFIG_CMD_MMC=y
+CONFIG_CMD_PART=y
+CONFIG_CMD_USB=y
+CONFIG_CMD_DHCP=y
+CONFIG_CMD_MII=y
+CONFIG_CMD_PING=y
+CONFIG_CMD_CACHE=y
+CONFIG_CMD_EXT2=y
+CONFIG_CMD_EXT4=y
+CONFIG_CMD_EXT4_WRITE=y
+CONFIG_CMD_FAT=y
+CONFIG_CMD_FS_GENERIC=y
+CONFIG_OF_CONTROL=y
+CONFIG_DEFAULT_DEVICE_TREE="r8a774b1-beacon-rzg2n-kit"
+CONFIG_ENV_IS_IN_MMC=y
+CONFIG_REGMAP=y
+CONFIG_SYSCON=y
+CONFIG_CLK=y
+CONFIG_CLK_RENESAS=y
+CONFIG_DM_GPIO=y
+CONFIG_RCAR_GPIO=y
+CONFIG_DM_PCA953X=y
+CONFIG_DM_I2C=y
+CONFIG_SYS_I2C_RCAR_I2C=y
+CONFIG_SYS_I2C_RCAR_IIC=y
+CONFIG_DM_MMC=y
+CONFIG_MMC_IO_VOLTAGE=y
+CONFIG_MMC_UHS_SUPPORT=y
+CONFIG_MMC_HS200_SUPPORT=y
+CONFIG_RENESAS_SDHI=y
+CONFIG_PHY_REALTEK=y
+CONFIG_DM_ETH=y
+CONFIG_RENESAS_RAVB=y
+CONFIG_PINCTRL=y
+CONFIG_PINCONF=y
+CONFIG_PINCTRL_PFC=y
+CONFIG_DM_REGULATOR=y
+CONFIG_DM_REGULATOR_FIXED=y
+CONFIG_DM_REGULATOR_GPIO=y
+CONFIG_SCIF_CONSOLE=y
+CONFIG_USB=y
+CONFIG_DM_USB=y
+CONFIG_WDT=y
+CONFIG_HW_WATCHDOG=y
+CONFIG_WDT_RENESAS=y
+CONFIG_USB_XHCI_HCD=y
+CONFIG_USB_EHCI_HCD=y
+CONFIG_USB_EHCI_GENERIC=y
+CONFIG_USB_STORAGE=y
+CONFIG_OF_LIBFDT_OVERLAY=y
+CONFIG_SMBIOS_MANUFACTURER=""
+CONFIG_PRAM=1
diff --git a/configs/r8a774e1_beacon-rzg2h_defconfig b/configs/r8a774e1_beacon-rzg2h_defconfig
new file mode 100644
index 0000000000..e4df7aeb17
--- /dev/null
+++ b/configs/r8a774e1_beacon-rzg2h_defconfig
@@ -0,0 +1,71 @@
+CONFIG_ARM=y
+CONFIG_ARCH_RMOBILE=y
+CONFIG_SYS_TEXT_BASE=0x50000000
+CONFIG_SYS_MALLOC_F_LEN=0x2000
+CONFIG_RCAR_GEN3=y
+CONFIG_R8A774E1=y
+CONFIG_TARGET_BEACON_RZG2H=y
+CONFIG_SMBIOS_PRODUCT_NAME=""
+CONFIG_FIT=y
+# CONFIG_ARCH_FIXUP_FDT_MEMORY is not set
+CONFIG_USE_BOOTARGS=y
+CONFIG_BOOTARGS="root=/dev/nfs rw nfsroot=192.168.0.1:/export/rfs ip=192.168.0.20"
+CONFIG_SUPPORT_RAW_INITRD=y
+CONFIG_DEFAULT_FDT_FILE="r8a774e1-beacon-rzg2h-kit.dtb"
+CONFIG_VERSION_VARIABLE=y
+CONFIG_HUSH_PARSER=y
+CONFIG_CMD_BOOTZ=y
+CONFIG_CMD_GPIO=y
+CONFIG_CMD_I2C=y
+CONFIG_CMD_MMC=y
+CONFIG_CMD_PART=y
+CONFIG_CMD_USB=y
+CONFIG_CMD_DHCP=y
+CONFIG_CMD_MII=y
+CONFIG_CMD_PING=y
+CONFIG_CMD_CACHE=y
+CONFIG_CMD_EXT2=y
+CONFIG_CMD_EXT4=y
+CONFIG_CMD_EXT4_WRITE=y
+CONFIG_CMD_FAT=y
+CONFIG_CMD_FS_GENERIC=y
+CONFIG_OF_CONTROL=y
+CONFIG_DEFAULT_DEVICE_TREE="r8a774e1-beacon-rzg2h-kit"
+CONFIG_ENV_IS_IN_MMC=y
+CONFIG_REGMAP=y
+CONFIG_SYSCON=y
+CONFIG_CLK=y
+CONFIG_CLK_RENESAS=y
+CONFIG_DM_GPIO=y
+CONFIG_RCAR_GPIO=y
+CONFIG_DM_PCA953X=y
+CONFIG_DM_I2C=y
+CONFIG_SYS_I2C_RCAR_I2C=y
+CONFIG_SYS_I2C_RCAR_IIC=y
+CONFIG_DM_MMC=y
+CONFIG_MMC_IO_VOLTAGE=y
+CONFIG_MMC_UHS_SUPPORT=y
+CONFIG_MMC_HS200_SUPPORT=y
+CONFIG_RENESAS_SDHI=y
+CONFIG_PHY_REALTEK=y
+CONFIG_DM_ETH=y
+CONFIG_RENESAS_RAVB=y
+CONFIG_PINCTRL=y
+CONFIG_PINCONF=y
+CONFIG_PINCTRL_PFC=y
+CONFIG_DM_REGULATOR=y
+CONFIG_DM_REGULATOR_FIXED=y
+CONFIG_DM_REGULATOR_GPIO=y
+CONFIG_SCIF_CONSOLE=y
+CONFIG_USB=y
+CONFIG_DM_USB=y
+CONFIG_WDT=y
+CONFIG_HW_WATCHDOG=y
+CONFIG_WDT_RENESAS=y
+CONFIG_USB_XHCI_HCD=y
+CONFIG_USB_EHCI_HCD=y
+CONFIG_USB_EHCI_GENERIC=y
+CONFIG_USB_STORAGE=y
+CONFIG_OF_LIBFDT_OVERLAY=y
+CONFIG_SMBIOS_MANUFACTURER=""
+CONFIG_PRAM=1
diff --git a/include/configs/beacon-rzg2m.h b/include/configs/beacon-rzg2.h
similarity index 100%
rename from include/configs/beacon-rzg2m.h
rename to include/configs/beacon-rzg2.h
-- 
2.17.1

